spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcWriter.java

```diff
@@ -237,9 +239,14 @@ public void addValue(int rowId, int column, SpecializedGetters data,
         output.noNulls = false;
         output.isNull[rowId] = true;
       } else {
+        Decimal decimal = data.getDecimal(column, precision, scale);
+        Preconditions.checkArgument(scale == decimal.scale(),
+            "Cannot write value as decimal(%s,%s), wrong scale: %s", precision, scale, decimal);
+        Preconditions.checkArgument(decimal.precision() <= precision && precision <= 18,
+            "Cannot write value as decimal(%s,%s), invalid precision: %s", decimal);
```

Comment:
This check seems redundant to me. If we are already passing our expected precision and scale to `data.getDecimal()`, wont the scale and precision of the returned decimal always match?

Reply:
Make sense.  they could be removed now.